pkg/block/index.go

```diff
@@ -358,38 +459,32 @@ func sanitizeChunkSequence(chks []chunks.Meta, mint int64, maxt int64) ([]chunks
 		return chks[i].MinTime < chks[j].MinTime
 	})
 
-	// Remove duplicates and complete outsiders.
+	// Remove duplicates, complete outsiders and near outsiders.
 	repl := make([]chunks.Meta, 0, len(chks))
-	for i, c := range chks {
-		if c.MinTime > maxt || c.MaxTime < mint {
-			// "Complete" outsider. Ignore.
-			continue
-		}
+	var last *chunks.Meta
+	for _, c := range chks {
+		var (
+			err    error
+			ignore = false
+		)
+
+		for _, ignoreChkFn := range ignoreChkFns {
+			ignore, err = ignoreChkFn(mint, maxt, last, &c)
+			if err != nil {
+				return nil, errors.Wrap(err, "ignore function")
+			}
 
-		if i == 0 {
-			repl = append(repl, c)
-			continue
+			if ignore {
+				break
```

Comment:
would labelling outer loop with `OUTER_LOOP:` and `continue OUTER_LOOP` work here?